lib/moodle_rb/courses.rb

```diff
@@ -70,20 +70,29 @@ def create(params)
     end
 
     def show(id)
+      if id.is_a?(Array)
+        ids = id.each_with_object({}) do |item, hash|
+          hash[id.index(item).to_s] = item
+        end
+      else
+        ids = { '0' => id }
+      end
```

Comment:
I'd recommend using a splat parameter (i.e. using `*id` as the parameter) instead and doing this:

```suggestion
    def show(*id)
      ids = id.each_with_index
              .to_h
              .invert
              .stringify_keys
```

Here's an example of this in action:
```
irb(main):068> id = [99, 44, 66]
=> [99, 44, 66]
irb(main):069> id.each_with_index.to_h.invert.stringify_keys
=> {"0"=>99, "1"=>44, "2"=>66}
```

Reply:
There might be use case of passing integer also. Does it work for integer also

Reply:
Also `stringify_keys` is not available in ruby 2.3

Reply:
> There might be use case of passing integer also. I don't think it will work?

Do you mean passing a single value to the `show` method? That is how splat is designed.

A method defined like so:
`def some_method(*params)`

Will have within it a `params` local variable that is _always_ an Array. 

These are both valid calling convention for the above:
* `some_method(12)`
* `some_method(12, 34, 54)`

You'll need to use the latter in your situation of calling multiple. If in Plus you have an array, you can splat the array in the call, and it will work the same:
```ruby
array_of_ids = [5, 6, 8]
some_method(*array_of_ids)
```